### PR TITLE
Auto fetching medium article

### PR DESCRIPTION
**Pull request by zquangu112z**
- Auto fetching medium article when:
    - Initializing app
    - Caching rate
- Add Makefile and docker-compose file for faster development

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import time
 from flask import Flask
 from flask_login import LoginManager
-
+import json
```
Comment (Naresh1318) [inline __init__.py]: Move this to line 2 for consistency

File: utils/medium.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import requests
+import json
+import logging
+from datetime import datetime
+
+
+def update_articles(db_conn, medium_url):
+    c = db_conn.cursor()
+    c.execute("DELETE FROM blogs")  # Clear all entries
```
Comment (Naresh1318) [inline utils/medium.py]: Clears manually added bogs each time the server starts. This is not desired. Try persisting the previous entries.

File: utils/medium.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    try:
+        retrieving_posts(medium_url, db_conn)
+        db_conn.commit()
+    except Exception as e:
+        logging.warning(f"[E0003] {e}")
+
+
+def retrieving_posts(medium_url, db_conn):
+    api = medium_url + "/latest?format=json"
```
Comment (Naresh1318) [inline utils/medium.py]: Python docstring missing for both functions

File: utils/medium.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    res = requests.get(api)
+    if not res:
+        raise Exception(
+            '''[E0001] An error occured when crawling data by page''')
+
+    content = json.loads(res.text.split('])}while(1);</x>')[1])
```
Comment (Naresh1318) [inline utils/medium.py]: Use single quotes for string to maintain consistancy

File: utils/database.py [hunk 5]
```diff
@@ -1,10 +1,10 @@
-import time
-import datetime
 from flask import current_app
 from flask_login import UserMixin
-
 from utils.github import update_public_repos
-
+from utils.medium import update_articles
+import datetime
```
Comment (Naresh1318) [inline utils/database.py]: Move inbuilt package imports to the top

File: utils/medium.py (hunk 2 above)
Comment (zquangu112z) [reply utils/medium.py]: I have updated as follow:

- Add new column `automatically_added` type INTEGER to table `blogs` to indicate if an entry is automatically added. 1 indicates True, 0 and NULL indicate False.
- When the function of updating blogs is triggered, it first deletes all the automatically added entries with `DELETE FROM blogs WHERE automatically_added = 1`, and then add crawled articles from Medium. So, manually added entries are kept.
- Update the code adding this column if missing when the app starts (@ db.py)

Comment (zquangu112z) [reply utils/medium.py]: The lasted commit is [here](https://github.com/Naresh1318/Topaz/pull/4/commits/5faa7fc17a40597f10733e4ebf61fe62268e32e5).

**Pull request merged**